core/platform/transactions/gae_transaction_services.py

```diff
@@ -48,6 +48,36 @@ def run_in_transaction(fn, *args, **kwargs):
     )
 
 
+def transaction_wrapper(fn, *args, **kwargs):
+    """Runs a function in a transaction. Either all of the operations in
+    the transaction are applied, or none of them are applied.
+
+    If an exception is raised, the transaction is likely not safe to
+    commit, since TransactionOptions.ALLOWED is used.
+
+    Args:
+        fn: callable. A function (or callable) to be called.
+        *args: list(*). Variable length argument list passed to the callable.
+        **kwargs: *. Arbitrary keyword arguments passed to the callable.
+
+    Returns:
+        *. Whatever fn() returns.
+
+    Raises:
+        Exception. Whatever fn() raises.
+        datastore_errors.TransactionFailedError. The transaction failed.
+    """
+    def wrapper():
+        """Wrapper for the transaction."""
+        return ndb.transaction(
+            lambda: fn(*args, **kwargs),
+            xg=True,
```

Comment:
What's this parameter for?

Reply:
Got it--maybe add a brief single-line comment explaining that or add to the documentation that this properly handles operations that involve multiple models.

Reply:
Done.